sw/device/silicon_creator/mask_rom/mask_rom.c

```diff
@@ -210,25 +341,41 @@ static rom_error_t mask_rom_boot(const manifest_t *manifest,
  * @return Result of the last attempt.
  */
 static rom_error_t mask_rom_try_boot(void) {
+  ROM_CFI_INCREMENT(kCfiRomTryBoot, 1);
+
   boot_policy_manifests_t manifests = boot_policy_manifests_get();
-  rom_error_t error = kErrorMaskRomBootFailed;
-  for (size_t i = 0; i < ARRAYSIZE(manifests.ordered); ++i) {
-    uint32_t flash_exec = 0;
-    error = mask_rom_verify(manifests.ordered[i], &flash_exec);
-    if (error != kErrorOk) {
-      continue;
-    }
-    // Boot fails if a verified ROM_EXT cannot be booted.
-    HARDENED_RETURN_IF_ERROR(mask_rom_boot(manifests.ordered[i], flash_exec));
-    // `mask_rom_boot()` should never return `kErrorOk`, but if it does
-    // we must shut down the chip instead of trying the next ROM_EXT.
+  uint32_t flash_exec = 0;
+
+  ROM_CFI_PREPCALL(kCfiRomTryBoot, 2, kCfiRomVerify);
+  rom_error_t error = mask_rom_verify(manifests.ordered[0], &flash_exec);
+  ROM_CFI_INCREMENT(kCfiRomTryBoot, 4);
+  ROM_CFI_CHECK(kCfiRomVerify, 3);
+
+  if (error == kErrorOk) {
+    HARDENED_CHECK_EQ(error, kErrorOk);
+    ROM_CFI_PREPCALL(kCfiRomTryBoot, 5, kCfiRomBoot);
+    HARDENED_RETURN_IF_ERROR(mask_rom_boot(manifests.ordered[0], flash_exec));
     return kErrorMaskRomBootFailed;
   }
-  return error;
+
+  ROM_CFI_INCREMENT(kCfiRomTryBoot, 5);
```

Comment:
IIUC the goal of this increment is to have different values for each branch. WDYT about removing this and adding a `ROM_CFI_INIT(kCfiRomTryBoot)` right before calling `mask_rom_boot()` in the if statement above?

Reply:
Updated to follow the suggestion. I think it is probably ok to use `ROM_CFI_INIT(kCfiRomTryBoot)` as the branch is not expected to return to the caller and `flash_exec` helps to provide additional protection.